install.hs

```diff
@@ -193,12 +198,18 @@ configureCabal versionNumber = do
 findInstalledGhcs :: IO [(VersionNumber, GhcPath)]
 findInstalledGhcs = do
   hieVersions <- getHieVersions :: IO [VersionNumber]
-  mapMaybeM
-    (\version -> getGhcPath version >>= \case
+  knownGhcs <- mapMaybeM
+    (\version -> getGhcPathOf version >>= \case
       Nothing -> return Nothing
       Just p  -> return $ Just (version, p)
     )
     (reverse hieVersions)
+  availableGhcs <- getGhcPaths
+  return
+    -- filter out stack provided GHCs
+    $ filter (not . isInfixOf ".stack" . snd)
```

Comment:
Why dont we want to have the possibility to use a ghc installed by stack?
I suppose, it contradicts your use-case, e.g. you can never install a custom ghc 8.6.4 if the stack-ghc is always on the path, right?

Reply:
> Why dont we want to have the possibility to use a ghc installed by stack?

Because this GHC finding logic is used only for cabal-related targets. If you'd like to use a stack-provided GHC, you'd use a hie-x.x.x target.

> I suppose, it contradicts your use-case, e.g. you can never install a custom ghc 8.6.4 if the stack-ghc is always on the path, right?

That's right. And I found a bug in my code. We should filter stack-provided GHCs _before_ applying `nub`. Otherwise a custom GHC 8.6.4 is always filtered out. I'll push a fix shortly.

Reply:
Pushed fdaf48b. We should squash it when merging this PR.